Embrace Unity iOS Interface/EmbraceManager.swift

```diff
@@ -0,0 +1,494 @@
+                                                "error.type": errorType
+                                            ],
+                                            events: [],
+                                            errorCode: .failure)
+    }
+    
+    static func startSpan(name: String, parentSpanId: String, startTimeMs: Double) -> String? {
+        let spanBuilder = Embrace.client?.buildSpan(name: name)
+        
+        guard let spanBuilder else {
+            return nil
+        }
+        
+        if !parentSpanId.isEmpty, let parent = spanRepository.get(spanId: parentSpanId) {
+            spanBuilder.setParent(parent)
+        } else {
+            spanBuilder.markAsKeySpan()
+        }
+        
+        if startTimeMs > 0.0 {
+            spanBuilder.setStartTime(time: convertDoubleToDate(ms: startTimeMs))
+        }
+        
+        let span = spanBuilder.startSpan()
+        
+        // Return the spanId
+        return spanRepository.spanStarted(span: span)
+    }
+    
+    static func stopSpan(spanId: String, errorCodeString: String, endTimeMs: Double) -> Bool {
+        guard let span = spanRepository.get(spanId: spanId) else {
+            return false
+        }
+        
+        if endTimeMs <= 0.0 {
+            span.end(errorCode: convertStringToErrorCode(str: errorCodeString))
+        } else {
+            span.end(errorCode: convertStringToErrorCode(str: errorCodeString),
+                      time: convertDoubleToDate(ms: endTimeMs))
+        }
+        
+        spanRepository.spanEnded(span: span)
+        
+        return true
+    }
+    
+    static func addSpanEventToSpan(spanId: String, name: String, time: Double, attributes: [String: AttributeValue]) -> Bool {
+        guard let span = spanRepository.get(spanId: spanId) else {
+            return false
+        }
+        
+        if attributes.isEmpty {
+            span.addEvent(name: name, timestamp: convertDoubleToDate(ms: time))
+        } else {
+            span.addEvent(name: name,
+                          attributes: attributes,
+                          timestamp: convertDoubleToDate(ms: time))
+        }
+        
+        return true
+    }
+    
+    static func addSpanAttributeToSpan(spanId: String, key: String, value: String) -> Bool {
+        guard let span = spanRepository.get(spanId: spanId) else {
+            return false
+        }
+        
+        span.setAttribute(key: key, value: value)
+        Embrace.client?.flush(span)
+        
+        return true
+    }
+    
+    static func recordCompletedSpan(
+        name: String,
+        startTimeMs: Double,
+        endTimeMs: Double,
+        errorCodeString: String,
+        parentSpanId: String,
+        attributes: inout [String: String],
+        events: [RecordingSpanEvent]) -> Bool {
+            
+            var parent = parentSpanId.isEmpty ? spanRepository.get(spanId: parentSpanId) : nil
+            attributes.updateValue("true", forKey: "emb.key")
```

Comment:
Using `EmbraceSemantics` would be:
```suggestion
            attributes.updateValue("true", forKey: SpanSemantics.keyIsKeySpan)
```